File: backend/app/schemas/social.py
from pydantic import BaseModel


class SocialPostResponse(BaseModel):
    id: str
    platform: str
    author: str
    content: str
    upvotes: int | None = None
    comments: int | None = None
    subreddit: str | None = None
    url: str
    published_at: str

File: frontend/src/types/social.ts
export type SocialPlatform = "reddit" | "twitter" | "linkedin";

export interface SocialPost {
  id: string;
  platform: SocialPlatform;
  author: string;
  content: string;
  upvotes?: number;
  comments?: number;
  subreddit?: string;
  url: string;
  publishedAt: string;
}
